main.py

```diff
@@ -1,45 +1,52 @@
+## Imports ##
 from math import floor
 
+## Functions ##
+def verify_input(user_input):
+    while True:
+        try:
+            user_input = int(input('''Please enter a valid date in the ddmmyyyy format
+    : '''))
+        except ValueError:
+            print("You did not enter a date.\n")
+            continue
+        user_input = str(user_input)
+        if len(user_input) != 8:
+            print('')
+            continue
+        if int(user_input[:2]) > 31:
+            print('')
+            continue
+        if int(user_input[3:4]) > 12:
+            print('')
+            continue
+        if int(user_input[4:]) < 1900:
+            print('\nYou must enter a year after 1900.')
+            continue
+        return
+
+def exit_loop():
+    while True:
+        exit_input = input('''\nTo exit this program, type exit
+: ''').lower()
+        if exit_input == 'exit':
+            print('\nExiting program...')
+            exit()
+        else:
+            continue
+
+## Main Code ##
 print("\nPasscode Generator\n")
 
-while True:
-    try:
-        user_input = int(input('''Please enter a valid date in the ddmmyyyy format
-: '''))
-    except ValueError:
-        print("You did not enter a date.\n")
-        continue
-    user_input = str(user_input)
-    if len(user_input) != 8:
-        print('')
-        continue
-    day = user_input[:2]
```

Comment:
You can't use string slicing on an integer - string is converted to an integer on line 7.